File: app/src/main/java/com/ynk/goodnews/adapters/AdapterListNews.java
package com.ynk.goodnews.adapters;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.RecyclerView;

import com.ynk.goodnews.R;
import com.ynk.goodnews.clicklisteners.AdapterItemClickListener;
import com.ynk.goodnews.databinding.NewsBinding;
import com.ynk.goodnews.model.News;

import java.util.List;

public class AdapterListNews extends RecyclerView.Adapter<AdapterListNews.NewsViewHolder> {

    private List<News> items;
    private AdapterItemClickListener adapterItemClickListener;

    public AdapterListNews(List<News> items, AdapterItemClickListener adapterItemClickListener) {
        this.items = items;
        this.adapterItemClickListener = adapterItemClickListener;
    }

    @Override
    public NewsViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        NewsBinding newsBinding = DataBindingUtil.inflate(LayoutInflater.from(parent.getContext()), R.layout.item_news_dashboard, parent, false);
        return new NewsViewHolder(newsBinding);
    }

    // Replace the contents of a view (invoked by the layout manager)
    @Override
    public void onBindViewHolder(NewsViewHolder holder, final int position) {
        holder.bind(getItem(position), adapterItemClickListener);
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    private News getItem(int position) {
        return items.get(position);
    }

    public class NewsViewHolder extends RecyclerView.ViewHolder {
        private NewsBinding newsBinding;

        public NewsViewHolder(NewsBinding newsBinding) {
            super(newsBinding.getRoot());
            this.newsBinding = newsBinding;
        }

        public void bind(News news, AdapterItemClickListener adapterItemClickListener) {
            this.newsBinding.setNews(news);
            this.newsBinding.setClickListener(adapterItemClickListener);
        }

    }

}

File: app/src/main/java/com/ynk/goodnews/adapters/AdapterListNews.kt
package com.ynk.goodnews.adapters

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import com.ynk.goodnews.R
import com.ynk.goodnews.clicklisteners.AdapterItemClickListener
import com.ynk.goodnews.databinding.NewsBinding
import com.ynk.goodnews.model.News

class AdapterListNews(
    private val items: List<News>,
    private val adapterItemClickListener: AdapterItemClickListener
) : RecyclerView.Adapter<AdapterListNews.NewsViewHolder2>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): NewsViewHolder2 {
        val newsBinding: NewsBinding = DataBindingUtil.inflate(
            LayoutInflater.from(parent.context),
            R.layout.item_news_dashboard,
            parent,
            false
        )
        return NewsViewHolder2(newsBinding)
    }

    override fun onBindViewHolder(holder: NewsViewHolder2, position: Int) {
        holder.bind(getItem(position), adapterItemClickListener)
    }

    override fun getItemCount() = items.size

    private fun getItem(position: Int) = items[position]

    inner class NewsViewHolder2(newsBinding: NewsBinding) : RecyclerView.ViewHolder(newsBinding.root) {
        private var newsBinding: NewsBinding = newsBinding

        fun bind(news: News?, adapterItemClickListener: AdapterItemClickListener?) {
            newsBinding.news = news
            newsBinding.clickListener = adapterItemClickListener
        }
    }
}